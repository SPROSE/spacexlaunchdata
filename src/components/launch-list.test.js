import React from 'react';
import { render } from '@testing-library/react';
import LaunchList from './launch-list';
import {shallow} from "enzyme/build/index";

const initialProps = {
        year: 0,
        spacexData: [],
        sort: 'ascending'
};

describe('<LaunchList /> with no spacexData', () => {
    const container = shallow(<LaunchList {...initialProps} />);
    it('should match the snapshot without launch data', () => {
        expect(container.html()).toMatchSnapshot();
    });
});

const newProps = {
    year: 0,
    spacexData: [{flight_number: 123,
        mission_name: 'Apollo',
        date: '2020-05-24T13:27:03+0000',
        rocket_name: 'Rocket-1',
        launch_year: '2009'}],
    sort: 'ascending'
};

describe('<LaunchList /> with spacexData', () => {
    const container = shallow(<LaunchList {...newProps} />);
    it('should match the snapshot with launch data', () => {
        expect(container.html()).toMatchSnapshot();
    });
});