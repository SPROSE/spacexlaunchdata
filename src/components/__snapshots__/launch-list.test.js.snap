// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchList /> with no spacexData should match the snapshot without launch data 1`] = `"<ul class=\\"launch-list\\"></ul>"`;

exports[`<LaunchList /> with spacexData should match the snapshot with launch data 1`] = `"<ul class=\\"launch-list\\"><li class=\\"launch-entry\\"><div class=\\"flight-no inline\\">#123</div><div class=\\"mission-name inline\\">Apollo</div><div class=\\"inline mid pull-right\\"><div class=\\"launch-date align-right\\">May 24, 2020</div><div class=\\"rocket-name align-right\\">Rocket-1</div></div></li></ul>"`;
